src/Plutarch/Extra/List.hs

```diff
@@ -1,380 +1,151 @@
-                xs <- pletC $ ptail # l
-
-                return $
-                    pif
-                        (eq # x # seen)
-                        (self # eq # seen # xs)
-                        (pcons # seen #$ self # eq # x # xs)
-
-{- | Special version of 'pnubSortBy', which requires elements have 'POrd' instance.
-
-   @since 1.1.0
--}
-pnubSort ::
-    forall (a :: S -> Type) (s :: S) list.
-    (PIsListLike list a, PIsListLike list (list a), POrd a) =>
-    Term s (list a :--> list a)
-pnubSort = phoistAcyclic $ pnubSortBy # eq # comp
-  where
-    eq = phoistAcyclic $ plam (#==)
-    comp = phoistAcyclic $ plam (#<)
-
-{- | / O(nlogn) /. Check if a list contains no duplicates.
-
-   @since 1.0.0
--}
-pisUniqBy ::
-    forall (a :: S -> Type) (s :: S) list.
-    (PIsListLike list a, PIsListLike list (list a)) =>
-    Term
-        s
-        ( (a :--> a :--> PBool)
-            :--> (a :--> a :--> PBool)
-            :--> list a
-            :--> PBool
-        )
-pisUniqBy = phoistAcyclic $
-    plam $ \eq comp xs ->
-        let nubbed = pnubSortBy # eq # comp # xs
-         in plength # xs #== plength # nubbed
-
-{- | A special case of 'pisUniqBy' which requires elements have 'POrd' instance.
-
-   @since 1.1.0
--}
-pisUniq ::
-    forall (a :: S -> Type) (s :: S) list.
-    (POrd a, PIsListLike list a, PIsListLike list (list a)) =>
-    Term s (list a :--> PBool)
-pisUniq = phoistAcyclic $ pisUniqBy # eq # comp
-  where
-    eq = phoistAcyclic $ plam (#==)
-    comp = phoistAcyclic $ plam (#<)
-
-{- | A special version of `pmap` which allows list elements to be thrown out.
-
-    @since 1.1.0
+ @since 3.4.0
 -}
 pmapMaybe ::
-    forall list (a :: S -> Type) (b :: S -> Type) (s :: S).
-    (PIsListLike list a, PIsListLike list b) =>
-    Term
-        s
-        ( (a :--> PMaybe b)
-            :--> list a
-            :--> list b
-        )
+    forall (b :: S -> Type) (ell :: (S -> Type) -> S -> Type) (a :: S -> Type) (s :: S).
+    (PElemConstraint ell a, PElemConstraint ell b, PListLike ell) =>
+    Term s ((a :--> PMaybe b) :--> ell a :--> ell b)
 pmapMaybe = phoistAcyclic $
-    pfix #$ plam $ \self f l -> pif (pnull # l) pnil $
-        unTermCont $ do
-            x <- pletC $ phead # l
-            xs <- pletC $ ptail # l
+    pfix #$ plam $ \self f xs ->
+        pmatch (PList.puncons # xs) $ \case
+            PNothing -> pnil
+            PJust xs' -> pmatch xs' $ \(PPair h t) ->
+                pmatch (f # h) $ \case
+                    PNothing -> self # f # t
+                    PJust h' -> pcons # h' # (self # f # t)
 
-            pure $
-                pmatch (f # x) $ \case
-                    PJust ux -> pcons # ux #$ self # f # xs
-                    _ -> self # f # xs
+{- | As 'pfind', but with a Haskell-level function on 'Term's for the predicate.
 
-{- | Get the first element that matches a predicate or return Nothing.
-
-     @since 1.1.0
+ @since 3.4.0
 -}
 pfind' ::
```

Comment:
We can really drop this and use `pfind`.

Reply:
Yeah, I'm not too sure why it's here either.